Travis script that runs engine unit tests and Dart tests of engine APIs

jason-simmons opened the pull request:
The Dart tests were migrated from the flutter/test/engine suite in the
framework repository

jason-simmons commented:
@abarth

Hixie commented:
![LGTM](http://junkyard.damowmow.com/566)

eseidelGoogle commented:
I don't speak bash well enough to know.  Will test.sh correctly exit on failure when any one of these lines fail?

Hixie commented on travis/test.sh:
```diff
@@ -0,0 +1,15 @@
+pub get
+popd
+
+for TEST_SCRIPT in flutter/testing/dart/*.dart; do
```
i recommend having a control test that verifies that if one of these fails, it properly gets noticed. See the framework test.sh for (a paranoidly high number of) examples.

abarth commented:
I'm not sure we actually compile enough of the engine on Travis to run the dart tests.

jason-simmons commented:
The set -e will exit after a failed command

Added verification of a test that fails intentionally.

Moved the script out of Travis with the intent of running it on the buildbots.

abarth commented:
![LGTM](http://i3.cpcache.com/product/1351564863/lgtm_bumper_stickers.jpg?width=200&height=200)
